### PR TITLE
[Merged by Bors] - Use new let-else syntax where possible

### PR DESCRIPTION
**Pull request by cart**
# Objective

Let-else syntax is now stable!

## Solution

Use it where possible!

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
Is there a lint for this yet? If so, can we enable it to ensure consistent style?

Do we want to merge this before automatic formatting is working?

**Comment by TimJentzsch** [comment]
> Is there a lint for this yet? If so, can we enable it to ensure consistent style?

[There is a lint for it](https://rust-lang.github.io/rust-clippy/master/index.html#manual_let_else), but it's only available on nightly so far.

File: crates/bevy_reflect/bevy_reflect_derive/src/container_attributes.rs [hunk 1]
```diff
@@ -172,12 +171,12 @@ impl ReflectTraits {
...
-                        segment.ident.to_string()
-                    } else {
+                    let Some(segment) = list.path.segments.iter().next() else {
                         continue;
                     };
 
+                    let ident = segment.ident.to_string();
+
                     // Track the span where the trait is implemented for future errors
                     let span = ident.span();
```
Comment (TimJentzsch) [inline crates/bevy_reflect/bevy_reflect_derive/src/container_attributes.rs:181]: Is this intended? In the code above the `.span()` is called on `segment.ident`, here it's called on `segment.ident.to_string()`.

I would also suggest to rename `ident` to `ident_name` here, to stay consistent with the code above.

Comment (IceSentry) [reply crates/bevy_reflect/bevy_reflect_derive/src/container_attributes.rs:181]: That's what the code did before too.

Comment (cart) [reply crates/bevy_reflect/bevy_reflect_derive/src/container_attributes.rs:181]: Yeah this was a minimal refactoring / thats what the name was before. But I'm happy to make targeted improvements and I agree this name could be better.

**Comment by cart** [comment]
> Do we want to merge this before automatic formatting is working?

I'm cool with letting people use the better syntax now with whatever looks ok and letting auto formatting work itself out, when it comes. The formatting discussion appears to have been going on for a long time and idk how long it will take to resolve itself. This way we don't accumulate "refactoring debt", whereas formatting style fixes will be automatic when they come.

**Review by IceSentry — APPROVED** [review]
The lack of clippy/fmt support is unfortunate, but I like the change and I'm 99% sure that's what the lint/fmt would end up looking like anyway.

**Review by alice-i-cecile — APPROVED** [review]

**Pull request closed without merging**